sdks/wasm/src/driver.c

```diff
@@ -674,3 +680,142 @@ mono_wasm_enable_on_demand_gc (void)
 {
 	mono_wasm_enable_gc = 1;
 }
+
+MonoAssembly* GetZoneInfoAssembly()
+{
+	static MonoAssembly* zoneInfoAssembly;
+
+	if (ZONE_INFO_FIRST_TIME) {
+
+		fprintf (stdout, "%s\n", "Locating ZoneInfo assembly.");
+		// try to open the ZoneInfo assembly
+		zoneInfoAssembly = mono_domain_assembly_open (root_domain, "WebAssembly.ZoneInfo");
+		// if not open then try to load it.
+		if (!zoneInfoAssembly) {
+			MonoImageOpenStatus status;
+			MonoAssemblyName* aname = mono_assembly_name_new ("WebAssembly.ZoneInfo");
+			zoneInfoAssembly = mono_assembly_load (aname, NULL, &status);
+			mono_assembly_name_free (aname);
+		}
+
+		if (zoneInfoAssembly)
+			fprintf (stdout, "%s\n", "ZoneInfo assembly located and loaded.");
+		else
+			fprintf (stdout, "%s\n", "ZoneInfo assembly could not be located.");
+
+		ZONE_INFO_FIRST_TIME = 0;
+	}
+
+    return zoneInfoAssembly;
+}
+
+MonoClass* GetZoneInfoClass()
+{
+	static MonoClass* klass;
+	if (!klass)
+		klass = mono_class_from_name(mono_assembly_get_image(GetZoneInfoAssembly()), "WebAssembly.ZoneInfo", "MonoWasmZoneInfo");
+	return klass;
```

Comment:
Kinda needs barriers.

Reply:
This is no longer relevant with current implementation.